Gems/Atom/RPI/Code/Source/RPI.Edit/Material/MaterialPropertyConnectionSerializer.cpp

```diff
@@ -0,0 +1,126 @@
+
+namespace AZ
+{
+    namespace RPI
+    {
+        namespace JsonMaterialPropertyConnectionSerializerInternal
+        {
+            namespace Field
+            {
+                static constexpr const char type[] = "type";
+                static constexpr const char name[] = "name";
+                static constexpr const char id[] = "id"; // For backward compatibility
+                static constexpr const char shaderIndex[] = "shaderIndex";
+            }
+
+            static const AZStd::string_view AcceptedFields[] =
+            {
+                Field::type,
+                Field::name,
+                Field::id,
+                Field::shaderIndex
+            };
+        }
+
+        AZ_CLASS_ALLOCATOR_IMPL(JsonMaterialPropertyConnectionSerializer, SystemAllocator, 0);
+
+        JsonSerializationResult::Result JsonMaterialPropertyConnectionSerializer::Load(void* outputValue, const Uuid& outputValueTypeId,
+            const rapidjson::Value& inputValue, JsonDeserializerContext& context)
+        {
+            namespace JSR = JsonSerializationResult;
+            using namespace JsonMaterialPropertyConnectionSerializerInternal;
+
+            AZ_Assert(azrtti_typeid<MaterialTypeSourceData::PropertyConnection>() == outputValueTypeId,
+                "Unable to deserialize material property connection to json because the provided type is %s",
+                outputValueTypeId.ToString<AZStd::string>().c_str());
+            AZ_UNUSED(outputValueTypeId);
+
+            MaterialTypeSourceData::PropertyConnection* propertyConnection = reinterpret_cast<MaterialTypeSourceData::PropertyConnection*>(outputValue);
+            AZ_Assert(propertyConnection, "Output value for JsonMaterialPropertyConnectionSerializer can't be null.");
+
+            JSR::ResultCode result(JSR::Tasks::ReadField);
+
+            if (!inputValue.IsObject())
+            {
+                return context.Report(JsonSerializationResult::Tasks::ReadField, JsonSerializationResult::Outcomes::Unsupported, "Property connection must be a JSON object.");
+            }
+            
+            MaterialUtils::CheckForUnrecognizedJsonFields(AcceptedFields, AZ_ARRAY_SIZE(AcceptedFields), inputValue, context, result);
+
+            result.Combine(ContinueLoadingFromJsonObjectField(&propertyConnection->m_type, azrtti_typeid<MaterialPropertyOutputType>(), inputValue, Field::type, context));
+            
+            JsonSerializationResult::ResultCode nameResult = ContinueLoadingFromJsonObjectField(&propertyConnection->m_fieldName, azrtti_typeid<AZStd::string>(), inputValue, Field::name, context);
+            if (nameResult.GetOutcome() == JsonSerializationResult::Outcomes::DefaultsUsed)
+            {
+                // This "id" key is for backward compatibility.
+                result.Combine(ContinueLoadingFromJsonObjectField(&propertyConnection->m_fieldName, azrtti_typeid<AZStd::string>(), inputValue, Field::id, context));
```

Comment:
Maybe `ContinueLoadingFromJsonObjectField` can be overloaded to check for multiple keys in order until one matches? Would allow providing "id" and "key" in one call.